fix(web): add next typegen to check-types script to prevent stale validator errors

## Problem

Running `pnpm run check-types` in `apps/web` fails with TS2307 errors from `.next/types/validator.ts`:

```
.next/types/validator.ts(1007,39): error TS2307: Cannot find module
'../../app/api/secrets/[name]/route.js' or its corresponding type declarations.
```

~20 routes report "Cannot find module" because the `.next/types/validator.ts` was generated by a previous `next build` when routes existed at different paths (e.g., `/api/secrets/` instead of `/api/zero/secrets/`). The stale build cache causes `tsc --noEmit` to fail.

## Root Cause

- `tsconfig.json` includes `.next/types/**/*.ts` (standard Next.js behavior)
- `validator.ts` is auto-generated by Next.js and is a snapshot of routes at generation time
- When routes are moved/renamed/deleted, the stale `validator.ts` references non-existent modules
- The current `check-types` script (`tsc --noEmit`) doesn't regenerate types before checking

## Solution

Change the `check-types` script in `apps/web/package.json` from:

```json
"check-types": "tsc --noEmit"
```

To:

```json
"check-types": "next typegen && tsc --noEmit"
```

`next typegen` (introduced in Next.js 15.5) generates fresh type files without running a full build. This is:
- The [official Next.js recommendation](https://nextjs.org/docs/app/api-reference/cli/next#next-typegen-options) for CI type-checking
- Already used in `apps/docs` in this monorepo
- Lightweight compared to `next build`

## References

- [Next.js docs: `next typegen`](https://nextjs.org/docs/app/api-reference/cli/next#next-typegen-options)
- [vercel/next.js#82916](https://github.com/vercel/next.js/issues/82916) — Docs update recommending `next typegen` for CI
- Deep dive research: `/tmp/deep-dive/next-types-check/`

--- a/turbo/apps/web/package.json
+++ b/turbo/apps/web/package.json
@@ -10,7 +10,7 @@
     "test": "vitest run",
     "test:watch": "vitest",
     "lint": "oxlint && eslint . --max-warnings 0",
-    "check-types": "tsc --noEmit",
+    "check-types": "next typegen && tsc --noEmit",
     "db:migrate": "tsx scripts/migrate.ts",
     "db:generate": "drizzle-kit generate",
     "db:dev-seed": "dotenv -e .env.local -- tsx scripts/dev-seed.ts",

